sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -39,54 +179,17 @@ def lambda_handler(event, context):
     """
     logger.info("Received event: " + json.dumps(event, indent=2))
 
-    logger.info(f"Event: {event}")
-    logger.info(f"Context: {context}")
     engine = db.get_engine()
 
-    # We're only expecting one record, but for some reason the Records are a list object
-    # TODO: events no longer have a Records key with list. This is already planned for
-    # removal in an upcoming PR.
-    for record in event["Records"]:
-        # Retrieve the Object name
-        logger.info(f"Record Received: {record}")
-        filename = record["s3"]["object"]["key"]
-
-        logger.info(f"Attempting to insert {os.path.basename(filename)} into database")
-        filename_parsed = FilenameParser(os.path.basename(filename))
-        filepath = filename_parsed.upload_filepath()
-
-        # confirm that the file is valid
-        if filepath["statusCode"] != 200:
-            logger.error(filepath["body"])
-            break
-
-        # setup a dictionary of metadata parameters to unpack in the
-        # instrument table
-        metadata_params = {
-            "file_path": filepath["body"],
-            "instrument": filename_parsed.instrument,
-            "data_level": filename_parsed.data_level,
-            "descriptor": filename_parsed.descriptor,
-            "start_date": datetime.datetime.strptime(
-                filename_parsed.startdate, "%Y%m%d"
-            ),
-            "end_date": datetime.datetime.strptime(filename_parsed.enddate, "%Y%m%d"),
-            "version": filename_parsed.version,
-            "extension": filename_parsed.extension,
-        }
-
-        # Add data to the file catalog
-        with Session(engine) as session:
-            session.add(models.FileCatalog(**metadata_params))
-            session.commit()
-
-            # TODO: These are sanity check. will remove
-            # from upcoming PR
-            result = session.query(models.FileCatalog).all()
-            for row in result:
-                print(row.instrument)
-                print(row.file_path)
-
-            inspector = inspect(engine)
-            table_names = inspector.get_table_names()
-            print(table_names)
+    # L0 data or other data type such as SPICE data ingestion
```

Comment:
I like how these are broken out into helper methods and dispatching out to the other handlers to  do the bulk of the work.